clang/tools/dpct/sycl-mapping-APIs-scanner.py

```diff
@@ -0,0 +1,225 @@
+
+"""
+A utility to help scan the sources of CUDA-SYCL migration projects and to provide a summary report listing unsupported APIs based on the support status of SYCL mapping/migration APIs tracked by Intel SYCLomatic open source project. The database of CUDA migration APIs status is stored under https://github.com/oneapi-src/SYCLomatic/tree/SYCLomatic/docs/dev_guide/api-mapping-status
+
+usage: sycl-mapping-APIs-scanner.py -p <cudaprojectfolder>
+
+"""
+import os
+import csv
+import sys
+#import re
+import api_migration_status_arrays
+
+
+def traverse_and_count_ASM_occurrences(folder_path, element_strings):
+    occur_dict = {}
+    for element in element_strings:
+        occur_dict[element] = 0
+        for root, dirs, files in os.walk(folder_path):
+            for file in files:
+                if file.endswith((".cu", ".cuh", ".cpp", ".c", ".hpp", ".h", ".py")):
+                    file_path = os.path.join(root, file)
+                    with open(file_path, 'r',errors='ignore') as file:
+                        contents = file.read()
+                        occur_dict[element] += contents.count(''.join(["\"", element, "."]))
+                        occur_dict[element] += contents.count(''.join(["\" ", element,"."]))
```

Comment:
For this match rule for ASM, it would match the "\${ASM_order}.".  It is rely on the Attributes of PTX instruction. @yihanwg yihan, can you confirm that the each PTX inst must have attributes when used?

Reply:
We usually ignore attributes when counting the migration status of a asm instruction.